### PR TITLE
Create admin authorization

### PR DESCRIPTION
**Pull request by drsoph**
- Added Edit/Delete option to become visible in Venue Index Container if user is an admin
- Re-wired React Router form component to render blank vs pre-filled form based on whether new form link or edit form link was clicked
- Venues api controller tests for update and delete features
@nwalberts

### CODE REVIEW

File: app/javascript/containers/VenueFormContainer.js [hunk 1]
```diff
@@ -36,88 +66,112 @@ class VenueFormContainer extends Component {
...
+    let url;
+    let method;
+
+    if(this.props.route.path == "/venues/:id/edit") {
+      url = `/api/v1/venues/${this.props.routeParams.id}`;
+      method = 'PATCH'
+    } else {
+      url = '/api/v1/venues';
+      method = 'POST'
+    }
```
Comment (nwalberts) [inline app/javascript/containers/VenueFormContainer.js]: Interesting, very interesting.

File: app/javascript/containers/VenuesIndexContainer.js [hunk 2]
```diff
@@ -8,12 +8,17 @@ class VenuesIndexContainer extends Component {
     super(props)
     this.state = {
       venues_array: [],
-      message: "hi cat"
+      message: "hi cat",
+      admin: false
     }
```
Comment (nwalberts) [inline app/javascript/containers/VenuesIndexContainer.js]: is `message` in state doing anything? Is this just hard coded?

File: app/models/user.rb [hunk 3]
```diff
@@ -7,9 +7,13 @@ class User < ApplicationRecord
...
+
   validates :first_name, presence: true
   validates :last_name, presence: true
   validates :email, presence: true
-  validates :password, presence: true
+
+  def admin?
+    role == "admin"
+    return true
+  end
```
Comment (nwalberts) [inline app/models/user.rb]: no need for line 17, this evaluates to true or false :P

Comment (drsoph) [reply app/models/user.rb]: our rspec test would evaluate to 'false' when we call admin.admin? unless we return true here for some reason :/

File: app/javascript/containers/VenueFormContainer.js (hunk 1 above)
Comment (drsoph) [reply app/javascript/containers/VenueFormContainer.js]: hope it's a good interesting!

**Pull request merged**